tests/components/nextcloud/test_coordinator.py

```diff
@@ -0,0 +1,73 @@
+"""Tests for the Nextcloud coordinator."""
+
+from unittest.mock import Mock, patch
+
+from freezegun.api import FrozenDateTimeFactory
+from nextcloudmonitor import (
+    NextcloudMonitor,
+    NextcloudMonitorAuthorizationError,
+    NextcloudMonitorConnectionError,
+    NextcloudMonitorError,
+    NextcloudMonitorRequestError,
+)
+import pytest
+
+from homeassistant.components.nextcloud.const import DEFAULT_SCAN_INTERVAL
+from homeassistant.components.nextcloud.coordinator import (
+    NextcloudDataUpdateCoordinator,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from . import mock_config_entry
+from .const import NC_DATA, VALID_CONFIG
+
+from tests.common import async_fire_time_changed
+
+
+@pytest.mark.parametrize(
+    ("error"),
+    [
+        (NextcloudMonitorAuthorizationError),
+        (NextcloudMonitorConnectionError),
+        (NextcloudMonitorRequestError),
+    ],
+)
+async def test_data_update(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory, error: NextcloudMonitorError
+) -> None:
+    """Test a coordinator data updates."""
+    entry = mock_config_entry(VALID_CONFIG)
+    entry.add_to_hass(hass)
+
+    with (
+        patch(
+            "homeassistant.components.nextcloud.NextcloudMonitor", spec=NextcloudMonitor
+        ) as mock_nextcloud_monitor,
+    ):
+        mock_nextcloud_monitor.return_value.update = Mock(
+            return_value=True,
+            side_effect=[None, error, None],
+        )
+        mock_nextcloud_monitor.return_value.data = NC_DATA
+        assert await hass.config_entries.async_setup(entry.entry_id)
+        coordinator: NextcloudDataUpdateCoordinator = entry.runtime_data
```

Comment:
Don't touch internals in tests. We should test the effect of this exception on the state machine

Reply:
And that would make this platform tests. I personally don't like `test_coordinator.py` because it assumes an internal mechanism, but we're testing behaviour